components/esp_modem/test/target_ota/components/manual_ota/transport_batch_tls.cpp

```diff
@@ -0,0 +1,232 @@
+        }
+        if (poll == 0) {
+            return ERR_TCP_TRANSPORT_CONNECTION_TIMEOUT;
+        }
+    }
+
+    auto ret =  tls->read(reinterpret_cast<unsigned char *>(buffer), len);
+    if (ret == MBEDTLS_ERR_SSL_WANT_READ) {
+        ret = ERR_TCP_TRANSPORT_CONNECTION_TIMEOUT;
+    }
+    ESP_LOGD(TAG, "red(len=%d, timeout=%d) ret=%d", len, timeout_ms, ret);
+    return ret;
+}
+
+int TlsTransport::transport::write(esp_transport_handle_t t, const char *buffer, int len, int timeout_ms)
+{
+    int poll;
+    if ((poll = esp_transport_poll_write(t, timeout_ms)) <= 0) {
+        ESP_LOGW(TAG, "Poll timeout or error timeout_ms=%d", timeout_ms);
+        return poll;
+    }
+
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    int ret = tls->write(reinterpret_cast<const unsigned char *>(buffer), len);
+    ESP_LOGD(TAG, "write ret=%d", ret);
+    return ret;
+}
+
+int TlsTransport::transport::close(esp_transport_handle_t t)
+{
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    int ret = esp_transport_close(tls->transport_);
+    tls->deinit();
+    return ret;
+}
+
+int TlsTransport::transport::poll_read(esp_transport_handle_t t, int timeout_ms)
+{
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    return esp_transport_poll_read(tls->transport_, timeout_ms);
+}
+
+int TlsTransport::transport::poll_write(esp_transport_handle_t t, int timeout_ms)
+{
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    return esp_transport_poll_write(tls->transport_, timeout_ms);
+}
+
+int TlsTransport::transport::destroy(esp_transport_handle_t t)
+{
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    return esp_transport_destroy(tls->transport_);
+}
+
+
+esp_transport_handle_t esp_transport_batch_tls_init(esp_transport_handle_t parent, const size_t max_buffer_size)
+{
+    esp_transport_handle_t ssl = esp_transport_init();
+    auto *tls = new TlsTransport(parent);
+    esp_transport_set_context_data(ssl, tls);
+    TlsTransport::set_func(ssl);
+    tls->prepare_buffer(max_buffer_size);
+    return ssl;
+}
+
+int TlsTransport::preread(size_t len, int timeout_ms)
+{
+    while (len != read_len) {
+        int l = esp_transport_read(transport_, buf.data() + read_len, len - read_len, timeout_ms);
+        ESP_LOGD(TAG, "need %d read %d already %d", len, l, read_len);
+        if ((l == ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN || l == ERR_TCP_TRANSPORT_CONNECTION_TIMEOUT ) && read_len > 0) {
+            return read_len;
+        }
+        if (l <= 0) {
+            read_len = 0;
+            return read_len;
+        }
+        read_len += l;
+    }
+    return read_len;
+}
+
+bool TlsTransport::prepare_buffer(size_t max_size)
+{
+    buf.resize(max_size);
+    return true;
+}
+
+int esp_transport_batch_tls_pre_read(esp_transport_handle_t t, int len, int timeout_ms)
```

Comment:
Should `int len` be changed to `size_t len` for consistency?

Reply:
good point, I'll change it to `size_t`